### DISCUSSION TITLE
Storing multiple programs in different program slots/numbers

### DISCUSSION THREAD

**Discussion by vijayaparthiban** [pybricks/support#1119] [General]
Hi, Not sure if I missed the most obvious documentation, but I could not find it anywhere.

Using: Spike Prime Hub, Pybricks beta.

As in the Spike Prime app (used code blocks so far), could we download a specific program to run when we press the programs '1', or '2', etc. For example, when program '1' is pressed, RUN this code.

**Comment by dlech** [comment]
Pybricks does not have this feature. Please see https://github.com/pybricks/support/issues/1064 for a similar feature request.

**Comment by vijayaparthiban** [comment]
Thanks. I think we need to use a different way of running multiple programs in robotics competitions then.

**Reply by laurensvalk** [reply]
Yes, you can already do it with Pybricks. See this example: https://github.com/pybricks/support/issues/861

**Comment by FLL-Team-24277** [comment]
@laurensvalk , we are planning on using a similar method to import different mission programs as needed. So we will have a BaseRobot class module, a master_program, and half-a-dozen or so mission programs. Do you have any thoughts on performance problems we might encounter? Right now it seems to work fine, but we only have one very small test mission, but it's just a matter of time before we start adding in other mission programs for imports. I hope the hub and firmware stay stable. 

I guess it is impossible to predict for sure if we are going to have problems without knowing how big each of the mission imports are, so let's just say each mission is twenty pybricks commands long, and let's say there are six of them.

You can see our base_robot, master_program, and sample mission clev_test in our repo
https://github.com/FLL-Team-24277/Master-Program-Fall-2023

I'm wondering if say, you have seen projects where someone has had a codebase ten times larger than what I am describing here, and they were able to use pybricks with no problems. That would be reassuring to know.

**Reply by laurensvalk** [reply]
There are a few interesting threads in this forum which might offer some inspiration, see e.g. https://github.com/orgs/pybricks/discussions/1062, which also includes a link to a team's code at the bottom.

> So we will have a BaseRobot class module, a master_program, and half-a-dozen or so mission programs. (...) so let's just say each mission is twenty pybricks commands long, and let's say there are six of them.

This should be fine. For comparison, people have solved Rubik's cubes or Connect4 games on the hub (even on hubs with less than half the capacity of SPIKE).

If you do run into issues, we're definitely interested to hear about it so we can fix it :)
